lib/semian/lru_cache.rb

```diff
@@ -0,0 +1,54 @@
+class LRUCache
+  extend Forwardable
+  def_delegators :@table, :size, :count
+  def_delegators :@table, :empty?
+  def_delegators :@table, :keys
+  def_delegators :@table, :clear
+
+  attr_accessor :table, :minimum_time_in_queue
+  def initialize(minimum_time_in_queue = 5)
+    @minimum_time_in_queue = minimum_time_in_queue * 60
+    @table = {}
+  end
+
+  def set(key, value)
+    @table.delete(key)
+    @table[key] = value
+    clear_unused_resources
+    value
+  end
+
+  def get(key)
+    if found = @table.delete(key)
+      @table[key] = found
+    end
+  end
+
+  def []=(key, value)
+    set(key, value)
+  end
+
+  def [](key)
+    get(key)
+  end
+
+  def delete(key)
+    @table.delete(key)
+  end
+
+  private
+
+  def clear_unused_resources
+    cleaned_elements = 0
+    @table.each do |key, value|
+      break if cleaned_elements >= 2
+      next if value.instance_of?(Semian::Resource)
+      next if value.circuit_breaker&.in_use?
+      next if value.created_at + minimum_time_in_queue > Time.now
```

Comment:
I'm not sure I understand what this heuristic does. For the first 5 minutes, it won't remove it even if it's empty? Why does that make a difference for an empty circuit?

Reply:
The problem I was getting is, for example, this test: https://github.com/Shopify/semian/blob/master/test/protected_resource_test.rb#L96

We register the resource, but at this point, there is no error in the window, which will cause the `clean_unused_resource` method to remove it as soon as it gets there. I feel like I might be missing something or maybe this should not happen.